core/modules/zmq_mdlw/include/klepsydra/zmq_core/zhelpers.hpp

```diff
@@ -73,31 +73,90 @@
 #endif
 
 //  Provide random number from 0..(num-1)
-#define within(num) (int) ((float) (num) * random () / (RAND_MAX + 1.0))
+#define within(num) (int) ((float)((num) * random ()) / (RAND_MAX + 1.0))
+
+//  Receive 0MQ string from socket and convert into C string
+//  Caller must free returned string.
+inline static char *
+s_recv(void *socket, int flags = 0) {
+	zmq_msg_t message;
```

Comment:
This code uses low-level zmq api syntax whereas the cppzmq api was used in the existing code. Better to stick to the high-level cppzmq api instead.

Reply:
cannot do it, I am afraid. This file comes from ZMQ and it is used by cppzmq. Gonna keep it that way for now....